ccl/changefeedccl: TestChangefeedBackfillCheckpoint failed

ccl/changefeedccl.TestChangefeedBackfillCheckpoint [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18922605?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18922605?buildTab=artifacts#/) on master @ [0fd08b1c5cc129c2b1714a81ded82248628d8603](https://github.com/cockroachdb/cockroach/commits/0fd08b1c5cc129c2b1714a81ded82248628d8603):


```
=== RUN   TestChangefeedBackfillCheckpoint
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/1b0af3c189d7f60ce633c41e08dd5486/logTestChangefeedBackfillCheckpoint3935595282
    test_log_scope.go:76: use -show-logs to present logs inline
    changefeed_test.go:7649: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/1b0af3c189d7f60ce633c41e08dd5486/logTestChangefeedBackfillCheckpoint3935595282
--- FAIL: TestChangefeedBackfillCheckpoint (24.27s)
=== RUN   TestChangefeedBackfillCheckpoint/webhook/limit=105_MB
    helpers_test.go:920: making server as system tenant
    helpers_test.go:996: making webhook feed factory
    changefeed_test.go:7612: 
        	Error Trace:	pkg/ccl/changefeedccl/changefeed_test.go:7612
        	            				pkg/ccl/changefeedccl/helpers_test.go:1203
        	Error:      	Should be true
        	Test:       	TestChangefeedBackfillCheckpoint/webhook/limit=105_MB
    --- FAIL: TestChangefeedBackfillCheckpoint/webhook/limit=105_MB (6.68s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/cdc
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestChangefeedBackfillCheckpoint.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-48094

--- a/pkg/ccl/changefeedccl/changefeed_test.go
+++ b/pkg/ccl/changefeedccl/changefeed_test.go
@@ -7578,6 +7578,12 @@ func TestChangefeedBackfillCheckpoint(t *testing.T) {
 			return false, nil
 		}
 
+		var actualFrontierStr atomic.Value
+		knobs.AfterCoordinatorFrontierRestore = func(frontier *resolvedspan.CoordinatorFrontier) {
+			require.NotNil(t, frontier)
+			actualFrontierStr.Store(frontier.String())
+		}
+
 		// Resume job.
 		require.NoError(t, jobFeed.Resume())
 
@@ -7589,14 +7595,14 @@ func TestChangefeedBackfillCheckpoint(t *testing.T) {
 			t.Fatal(err)
 		}
 		assert.NoError(t, checkpoint.Restore(expectedFrontier, timestampSpansMap))
-
-		var checkedCoordinatorFrontier atomic.Bool
-
-		knobs.AfterCoordinatorFrontierRestore = func(frontier *resolvedspan.CoordinatorFrontier) {
-			require.NotNil(t, frontier)
-			require.Equal(t, expectedFrontier.String(), frontier.String())
-			checkedCoordinatorFrontier.Store(true)
-		}
+		expectedFrontierStr := expectedFrontier.String()
+		testutils.SucceedsSoon(t, func() error {
+			if s := actualFrontierStr.Load(); s != nil {
+				require.Equal(t, expectedFrontierStr, s)
+				return nil
+			}
+			return errors.New("waiting for frontier to be restored")
+		})
 
 		// Wait for the high water mark to be non-zero.
 		testutils.SucceedsSoon(t, func() error {
@@ -7606,11 +7612,6 @@ func TestChangefeedBackfillCheckpoint(t *testing.T) {
 			}
 			return errors.New("waiting for highwater")
 		})
-
-		// Now that we've checked that our checkpoint progress is loaded on resume,
-		// do not make further checks.
-		require.True(t, checkedCoordinatorFrontier.Load())
-		knobs.AfterCoordinatorFrontierRestore = nil
 
 		// At this point, highwater mark should be set, and previous checkpoint should be gone.
 		progress = loadProgress()
@@ -7644,7 +7645,7 @@ func TestChangefeedBackfillCheckpoint(t *testing.T) {
 	// TODO(ssd): Tenant testing disabled because of use of DB()
 	for _, sz := range []int64{100 << 20, 100} {
 		maxCheckpointSize = sz
-		cdcTestNamedWithSystem(t, fmt.Sprintf("limit=%s", humanize.Bytes(uint64(sz))), testFn, feedTestForceSink("webhook"))
+		cdcTestNamedWithSystem(t, fmt.Sprintf("limit=%s", humanize.Bytes(uint64(sz))), testFn, feedTestEnterpriseSinks)
 	}
 }
 

